src/views/common/Animation/AnimationControls/usePlaybackBarGeometry.tsx

```diff
@@ -0,0 +1,81 @@
+import { norm } from 'mathjs'
```

Comment:
This file breaks out (and memoizes) most of the stuff that describes the scrub bar based purely on its geometry. The idea is a) to make that more straightforward and b) to avoid rerunning the parts of hooks that don't need to be rerun just because the current time or displayed-frame changed--we don't need to recompute the padding on the bars and all that just because we changed what frame is up.